hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializerTest.java

```diff
@@ -581,34 +694,84 @@ public Class<EmployeeDTO> getCompactClass() {
         //Using this registration to mimic schema evolution. This is usage is not advised.
-        serializationConfig.getCompactSerializationConfig()
-                .addSerializer(new CompactSerializer<EmployeeDTO>() {
-                    @Nonnull
-                    @Override
-                    public EmployeeDTO read(@Nonnull CompactReader reader) {
-                        throw new UnsupportedOperationException("We will not read from here on this test");
-                    }
+        serializationConfig.getCompactSerializationConfig().addSerializer(new CompactSerializer<EmployeeDTO>() {
+            @Nonnull
+            @Override
+            public EmployeeDTO read(@Nonnull CompactReader in) {
+                throw new UnsupportedOperationException("We will not read from here on this test");
+            }
+
+            @Override
+            public void write(@Nonnull CompactWriter out, @Nonnull EmployeeDTO object) {
+                out.writeInt32("age", object.getAge());
+                out.writeInt64("id", object.getId());
+                out.writeInt8("height", (byte) 187);
+            }
+
+            @Nonnull
+            @Override
+            public String getTypeName() {
+                return EmployeeDTO.class.getName();
+            }
+
+            @Nonnull
+            @Override
+            public Class<EmployeeDTO> getCompactClass() {
+                return EmployeeDTO.class;
+            }
+        });
 
-                    @Override
-                    public void write(@Nonnull CompactWriter writer, @Nonnull EmployeeDTO object) {
-                        writer.writeInt32("age", object.getAge());
-                    }
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setConfig(serializationConfig)
+                .setSchemaService(schemaService)
+                .build();
 
-                    @Nonnull
-                    @Override
-                    public String getTypeName() {
-                        return EmployeeDTO.class.getName();
-                    }
+        EmployeeDTO expected = new EmployeeDTO(20, 102310312);
+        Data data = serializationService.toData(expected);
 
-                    @Nonnull
-                    @Override
-                    public Class<EmployeeDTO> getCompactClass() {
-                        return EmployeeDTO.class;
-                    }
-                });
+        SerializationConfig serializationConfig2 = new SerializationConfig();
+        SerializationService serializationService2 = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(serializationConfig2)
+                .build();
+
+        EmployeeDTO actual = serializationService2.toObject(data);
+
+        assertEquals(expected.getAge(), actual.getAge());
+        assertEquals(expected.getId(), actual.getId());
```

Comment:
Should we test the other way around as well? (reading the data written by the old writer on the new reader) 

Similar for other tests as well

Reply:
Can you check `testSchemaEvolution_variableSizeFieldAdded` ? I added as you suggested. If it's ok, I will add it to other schema evol tests as well.